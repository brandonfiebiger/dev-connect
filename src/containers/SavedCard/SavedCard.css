.saved-card {
  background-color: #55575c;
  border-radius: 25px;
  opacity: 0.95;
  width: 700px; }
