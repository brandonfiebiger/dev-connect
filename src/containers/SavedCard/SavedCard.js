import React, { Component } from 'react';
import { connect } from 'react-redux';
import { removeJob } from '../../actions';
import './SavedCard.css';

export class SavedCard extends Component {
  updateJobSaveStatus = async () => {
    const { id } = this.props;

    const url = process.env.REACT_APP_DATABASE_API_URL + `/api/v1/jobs/${id}`;
    try {
      const response = await fetch(url, {
        method: 'PATCH',
        body: JSON.stringify({
          status: 'none'
        }),
        headers: {
          'content-type': 'application/json'
        }
      });

      await response.json();
    } catch (error) {
      console.log('something went wrong', error);
    }

    this.handleRemoveCard();
  };

  handleRemoveCard = () => {
    const { removeJob, id } = this.props;

    removeJob(id);
  };

  render() {
    const { description, company, location, title } = this.props;

    return (
      <div className="job-card">
        <h3>{company}</h3>
        <h4>{title}</h4>
        <p>{location}</p>
        <p>{description}</p>
        <button onClick={this.updateJobSaveStatus}>Remove Job</button>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  removeJob: job => dispatch(removeJob(job))
});

export default connect(
  null,
  mapDispatchToProps
)(SavedCard);
