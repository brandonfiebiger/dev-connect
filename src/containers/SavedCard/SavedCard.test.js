import React from 'react';
import { shallow } from 'enzyme';
import { SavedCard } from './SavedCard';

describe('SavedCard', () => {
  let wrapper;
  let mockCompany = 'Gusto';
  let mockLocation = 'Denver, CO';
  let mockType = 'UX Designer';
  let mockStatus = 'none';
  let mockDescription = 'We are looking for a talented developer';

  beforeEach(() => {
    wrapper = shallow(
      <SavedCard
        description={mockDescription}
        company={mockCompany}
        location={mockLocation}
        title={mockType}
        status={mockStatus}
        id={55}
        removeJob={jest.fn()}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
