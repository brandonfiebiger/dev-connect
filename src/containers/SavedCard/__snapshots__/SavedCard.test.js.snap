// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SavedCard
    company="Gusto"
    description="We are looking for a talented developer"
    id={55}
    location="Denver, CO"
    removeJob={[MockFunction]}
    status="none"
    title="UX Designer"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Gusto
        </h3>,
        <h4>
          UX Designer
        </h4>,
        <p>
          Denver, CO
        </p>,
        <p>
          We are looking for a talented developer
        </p>,
        <button
          onClick={[Function]}
        >
          Remove Job
        </button>,
      ],
      "className": "job-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Gusto",
        },
        "ref": null,
        "rendered": "Gusto",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "UX Designer",
        },
        "ref": null,
        "rendered": "UX Designer",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Denver, CO",
        },
        "ref": null,
        "rendered": "Denver, CO",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "We are looking for a talented developer",
        },
        "ref": null,
        "rendered": "We are looking for a talented developer",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remove Job",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove Job",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Gusto
          </h3>,
          <h4>
            UX Designer
          </h4>,
          <p>
            Denver, CO
          </p>,
          <p>
            We are looking for a talented developer
          </p>,
          <button
            onClick={[Function]}
          >
            Remove Job
          </button>,
        ],
        "className": "job-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Gusto",
          },
          "ref": null,
          "rendered": "Gusto",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "UX Designer",
          },
          "ref": null,
          "rendered": "UX Designer",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Denver, CO",
          },
          "ref": null,
          "rendered": "Denver, CO",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "We are looking for a talented developer",
          },
          "ref": null,
          "rendered": "We are looking for a talented developer",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remove Job",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove Job",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
